tree.c

```diff
@@ -178,26 +198,37 @@ next:		;
 		}
 
 		printf("%s-- %s", prefix, ents[i].name);
-		if (Fflag)
-			print_suffix(ents[i].mode);
-		printf("\n");
-
-		if (S_ISDIR(ents[i].mode)) {
-			char *subdir, *subprefix;
-
-			if (asprintf(&subdir, "%s/%s", dir,
-			    ents[i].name) == -1)
-				err(1, "asprintf");
-			if (asprintf(&subprefix, "%s   |", prefix) == -1)
-				err(1, "asprintf");
-			if (i == n - 1) {
-				subprefix[prefix_len - 1] = ' ';
+
+		if (S_ISLNK(ents[i].mode)) {
+			char buf[PATH_MAX];
+			ssize_t nbytes;
+			nbytes = readlink(ents[i].path, buf, PATH_MAX);
+			if (nbytes < 1)
+				err(1, "readlink");
+			printf(" -> %.*s\n", (int) nbytes, buf);
```

Comment:
This actually needs to print the mode on the target, as in:

```
foo/
`-- bar -> baz/
```

But i need to store the mode for that, too lazy rn